File: Downloads/Esprit-PIDEV-3A20-2526-Nexora-integration/src/Entity/OptionVariation.php
<?php

namespace App\Entity;

use App\Repository\OptionVariationRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: OptionVariationRepository::class)]
#[ORM\Table(name: 'option_variation')]
class OptionVariation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(type: 'integer')]
    private int $attributId = 0;

    #[ORM\ManyToOne(targetEntity: AttributVariation::class, inversedBy: 'options')]
    #[ORM\JoinColumn(name: 'attribut_id', referencedColumnName: 'id', nullable: true)]
    private ?AttributVariation $attribut = null;

    #[ORM\Column(length: 100)]
    private string $valeur = '';

    #[ORM\Column(nullable: true, length: 7)]
    private ?string $codeHexadecimal = null;

    #[ORM\Column(type: 'integer', nullable: true)]
    private ?int $ordreAffichage = null;

    public function getId(): ?int { return $this->id; }

    public function getAttributId(): int { return $this->attributId; }
    public function setAttributId(int $attributId): static { $this->attributId = $attributId; return $this; }

    public function getAttribut(): ?AttributVariation { return $this->attribut; }
    public function setAttribut(?AttributVariation $attribut): static
    {
        $this->attribut = $attribut;
        if ($attribut) $this->attributId = $attribut->getId() ?? 0;
        return $this;
    }

    public function getValeur(): string { return $this->valeur; }
    public function setValeur(string $valeur): static { $this->valeur = $valeur; return $this; }

    public function getCodeHexadecimal(): ?string { return $this->codeHexadecimal; }
    public function setCodeHexadecimal(?string $codeHexadecimal): static { $this->codeHexadecimal = $codeHexadecimal; return $this; }

    public function getOrdreAffichage(): ?int { return $this->ordreAffichage; }
    public function setOrdreAffichage(?int $ordreAffichage): static { $this->ordreAffichage = $ordreAffichage; return $this; }
}

File: nexora_java/src/main/java/com/pi/entities/OptionVariation.java
package com.pi.entities;

public class OptionVariation {
    private int id;
    private int attributId;
    private String valeur;
    private String codeHexadecimal; // Pour les couleurs
    private int ordreAffichage;

    // Constructeurs
    public OptionVariation() {}

    public OptionVariation(int attributId, String valeur) {
        this.attributId = attributId;
        this.valeur = valeur;
    }

    // Getters et Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getAttributId() { return attributId; }
    public void setAttributId(int attributId) { this.attributId = attributId; }

    public String getValeur() { return valeur; }
    public void setValeur(String valeur) { this.valeur = valeur; }

    public String getCodeHexadecimal() { return codeHexadecimal; }
    public void setCodeHexadecimal(String codeHexadecimal) { this.codeHexadecimal = codeHexadecimal; }

    public int getOrdreAffichage() { return ordreAffichage; }
    public void setOrdreAffichage(int ordreAffichage) { this.ordreAffichage = ordreAffichage; }

    @Override
    public String toString() {
        return valeur;
    }
}
